langlib/langlib-test/src/test/resources/test-src/floatlib_test.bal

```diff
@@ -415,6 +415,720 @@ function testFromStringNegative() {
+    float a = 5.7;
+
+    string|error b = trap a.toFixedString(-2);
+    assertEquality(true, b is error);
+    if (b is error) {
+        assertEquality("{ballerina/lang.float}InvalidFractionDigits", b.message());
+        assertEquality("fraction digits cannot be less than 0", <string> checkpanic b.detail()["message"]);
+    }
+
+    b = trap float:toFixedString(a, -2);
+    assertEquality(true, b is error);
+    if (b is error) {
+        assertEquality("{ballerina/lang.float}InvalidFractionDigits", b.message());
+        assertEquality("fraction digits cannot be less than 0", <string> checkpanic b.detail()["message"]);
+    }
+
+    b = trap float:toFixedString(a, -(+2));
+    assertEquality(true, b is error);
+    if (b is error) {
+        assertEquality("{ballerina/lang.float}InvalidFractionDigits", b.message());
+        assertEquality("fraction digits cannot be less than 0", <string> checkpanic b.detail()["message"]);
+    }
+
+    b = trap float:toFixedString(a, ~2);
+    assertEquality(true, b is error);
+    if (b is error) {
+        assertEquality("{ballerina/lang.float}InvalidFractionDigits", b.message());
+        assertEquality("fraction digits cannot be less than 0", <string> checkpanic b.detail()["message"]);
+    }
+}
+
+function testToFixedStringWhenFractionDigitsIsZero() {
+    float a = 5.7;
+    string b = float:toFixedString(a, 0);
+    assertEquality("6", b);
+    assertEquality("6", a.toFixedString(0));
+
+    float c = -5.7;
+    string d = float:toFixedString(c, 0);
+    assertEquality("-6", d);
+    assertEquality("-6", c.toFixedString(0));
+}
+
+function testToFixedStringWhenFractionDigitsIsNil() {
+    float a = 5.7;
+    string b = float:toFixedString(a, ());
+    assertEquality("5.7", b);
+    assertEquality("5.7", a.toFixedString(()));
+
+    float c = -5.7;
+    string d = float:toFixedString(c, ());
+    assertEquality("-5.7", d);
+    assertEquality("-5.7", c.toFixedString(()));
+}
+
+function testToFixedStringWhenFractionDigitsIsVeryLargeInt() {
+    float a = 5.7;
+    string b = float:toFixedString(a, 2147483648);
+    assertEquality("5.7", b);
+    assertEquality("5.7", a.toFixedString(2147483647 + 1));
+
+    float c = -5.7;
+    string d = float:toFixedString(c, 2147483648 + 1);
+    assertEquality("-5.7", d);
+    assertEquality("-5.7", c.toFixedString(2147483647 + 1));
+}
+
+function testToFixedStringWhenFractionDigitsIsIntMax() {
+    float a = 5.7;
+    string b = float:toFixedString(a, 9223372036854775807);
+    assertEquality("5.7", b);
+    assertEquality("5.7", a.toFixedString(int:MAX_VALUE));
+
+    float c = -5.7;
+    string d = float:toFixedString(c, 9223372036854775807);
+    assertEquality("-5.7", d);
+    assertEquality("-5.7", c.toFixedString(int:MAX_VALUE));
+}
+
+function testToFixedStringWithMorePositiveFloats() {
+    assertEquality("45362.1233399999982794", float:toFixedString(45362.12334, 16));
+    assertEquality("0.1237469219000000", float:toFixedString(0.1237469219, 16));
```

Comment:
Shall we add one or more test case to reflect rounding mode in `float:toFixedString`